javascript/javascript3/week1/main/functions.js

```diff
@@ -0,0 +1,286 @@
+        });
+      }
+    }, 1000);
+    console.log(cardsOpened);
+  } else {
+    cardsOpened.push(cardOnDesk.id);
+    console.log(cardsOpened);
+  }
+}
+
+function pushingCardsToDesk(amountOfCards, cards) {
+  //pushing some amount of cards to desk-array and doubling them (that's why I'm pushing this cards two times)
+  for (let i = 0; i < amountOfCards; i++) {
+    desk.push(cards[i]);
+    desk.push(cards[i]);
+  }
+}
+
+function startTimer(count) {
+  //starting timer
+  if (count === 0) {
+    const timerForSeconds = setInterval(function () {
+      if (seconds > 58) {
+        seconds = 0;
+        timerSeconds.innerText = "00";
+      } else if (seconds < 9) {
+        seconds++;
+        timerSeconds.innerText = "0" + seconds;
+      } else {
+        seconds++;
+        timerSeconds.innerText = seconds;
+      }
+    }, 1000);
+    const timerForMinutes = setInterval(function () {
+      if (minutes < 9) {
+        minutes++;
+        timerMinutes.innerText = "0" + minutes;
+      } else {
+        minutes++;
+        timerMinutes.innerText = minutes;
+      }
+    }, 60000);
+    if (amountOfCardsOnTheDesk <= 0) {
+      clearInterval(timerForMinutes);
+      clearInterval(timerForSeconds);
+    }
+  }
+}
+
+//function, that counts amount of flips
+function countFlipping() {
+  count++;
+  countText.innerText = count;
+}
+
+//function for ending of game
+function endGame() {
+  desk.length = 0;
+  endingMenu.removeAttribute("style");
+  countContainer.setAttribute("style", "display: none;");
+}
+
+//function for restarting the game
+function restartingGame() {
+  count = 0;
+  countText.innerText = count;
+  seconds = 0;
+  minutes = 0;
+  endingMenu.setAttribute("style", "display: none;");
+  startingMenu.removeAttribute("style");
+  countContainer.setAttribute("style", "display: none;");
+}
+
+
+
+
+
+
+
+//EasterEgg function (I will draw a bunch of cards... Again)
+function setEasterDesk() {
+  const password = document.getElementById("password").value;
+
+  if (password === "3336633") {
+    settingEasterDesk();
+  }
+}
+
+function settingEasterDesk() {
+  count = 0;
+  countText.innerText = count;
+  seconds = 0;
+  minutes = 0;
+
+  endingMenu.setAttribute("style", "display:none;");
+  easterMenu.setAttribute("style", "display:none;");
+  amountOfCardsOnTheDesk = easterCards.length;
+  //showing container with count
+  countContainer.removeAttribute("style");
+
+  pushingCardsToDesk(easterCards.length, easterCards);
+
+  //shuffling cards
+  const sortedDesk = desk.sort(() => {
+    return 0.5 - Math.random();
+  });
+
+  //creating cards, setting attributes and setting event listener
+  sortedDesk.forEach((card) => {
+    const deskCard = document.createElement("div");
+    //deskCard.setAttribute("title", card.name);
+    deskCard.setAttribute("class", "card");
+    deskCard.setAttribute("id", card.id);
+    deskMenu.appendChild(deskCard);
+
+    deskCard.addEventListener("click", () => {
```

Comment:
As I mentioned in the previous PR, this code is a duplication of the lines 67 to 100. You need to use functions to share code between multiple places.